### PR TITLE
stack list-dependencies:  use --system-ghc

### PR DESCRIPTION
**Pull request by deepfire**
Tested to generate a nix expression sufficiently good to start building `cardano-sl`.

### CODE REVIEW

File: stack2nix/Main.hs [hunk 1]
```diff
@@ -9,6 +9,7 @@ args = Args
        <$> optional (strOption $ long "revision" <> help "revision to use when fetching from VCS")
        <*> optional (strOption $ short 'o' <> help "output file for generated nix expression")
        <*> strArgument (metavar "URI")
+       <*> (flag False True $ long "system-ghc" <> help "pass --system-ghc to stack invocations")
```
Comment (jmitchell) [inline stack2nix/Main.hs]: Please reorder this so the `URI` comes last. It makes `stack2nix --help` more readable.

Otherwise, looks great! Thanks 👍

Comment (deepfire) [inline stack2nix/Main.hs]: Done!

Comment (jmitchell) [inline stack2nix/Main.hs]: The record fields need to be reordered to match.

Comment (deepfire) [inline stack2nix/Main.hs]: Oh, :blush:..

**Pull request merged**